format/moc3/moc3.go

```diff
@@ -0,0 +1,1766 @@
+			countInfo.artMeshes = int64(d.FieldU32("art_meshes"))
+			countInfo.parameters = int64(d.FieldU32("parameters"))
+			countInfo.partKeyforms = int64(d.FieldU32("part_keyforms"))
+			countInfo.warpDeformerKeyforms = int64(d.FieldU32("warp_deformer_keyforms"))
+			countInfo.rotationDeformerKeyforms = int64(d.FieldU32("rotation_deformer_keyforms"))
+			countInfo.artMeshKeyforms = int64(d.FieldU32("art_mesh_keyforms"))
+			countInfo.keyformPositions = int64(d.FieldU32("keyform_positions"))
+			countInfo.parameterBindingIndices = int64(d.FieldU32("parameter_binding_indices"))
+			countInfo.keyformBindings = int64(d.FieldU32("keyform_bindings"))
+			countInfo.parameterBindings = int64(d.FieldU32("parameter_bindings"))
+			countInfo.keys = int64(d.FieldU32("keys"))
+			countInfo.uvs = int64(d.FieldU32("uvs"))
+			countInfo.positionIndices = int64(d.FieldU32("position_indices"))
+			countInfo.drawableMasks = int64(d.FieldU32("drawable_masks"))
+			countInfo.drawOrderGroups = int64(d.FieldU32("draw_order_groups"))
+			countInfo.drawOrderGroupObjects = int64(d.FieldU32("draw_order_group_objects"))
+			countInfo.glue = int64(d.FieldU32("glue"))
+			countInfo.glueInfo = int64(d.FieldU32("glue_info"))
+
+			if version < moc3Version4_02_00 {
+				return
+			}
+
+			countInfo.glueKeyforms = int64(d.FieldU32("glue_keyforms"))
+			countInfo.keyformMultiplyColors = int64(d.FieldU32("keyform_multiply_colors"))
+			countInfo.keyformScreenColors = int64(d.FieldU32("keyform_screen_colors"))
+			countInfo.blendShapeParameterBindings = int64(d.FieldU32("blend_shape_parameter_bindings"))
+			countInfo.blendShapeKeyformBindings = int64(d.FieldU32("blend_shape_keyform_bindings"))
+			countInfo.blendShapesWarpDeformers = int64(d.FieldU32("blend_shapes_warp_deformers"))
+			countInfo.blendShapesArtMeshes = int64(d.FieldU32("blend_shapes_art_meshes"))
+			countInfo.blendShapeConstraintIndices = int64(d.FieldU32("blend_shape_constraint_indices"))
+			countInfo.blendShapeConstraints = int64(d.FieldU32("blend_shape_constraints"))
+			countInfo.blendShapeConstraintValues = int64(d.FieldU32("blend_shape_constraint_values"))
+
+			if version < moc3Version5_00_00 {
+				return
+			}
+
+			countInfo.blendShapesParts = int64(d.FieldU32("blend_shapes_parts"))
+			countInfo.blendShapesRotationDeformers = int64(d.FieldU32("blend_shapes_rotation_deformers"))
+			countInfo.blendShapesGlue = int64(d.FieldU32("blend_shapes_glue"))
+		})
+
+		d.SeekAbs(sectionOffsets.canvasInfo * 8)
+		d.FieldStruct("canvas_info", func(d *decode.D) {
+			d.FieldF32("pixels_per_unit")
+			d.FieldF32("origin_x")
+			d.FieldF32("origin_y")
+			d.FieldF32("canvas_width")
+			d.FieldF32("canvas_height")
+			d.FieldU8("canvas_flags")
+		})
+
+		d.FieldStruct("parts", func(d *decode.D) {
+			d.SeekAbs(sectionOffsets.parts.ids * 8)
+			d.FieldArray("ids", func(d *decode.D) {
+				for i := int64(0); i < countInfo.parts; i++ {
+					d.FieldUTF8NullFixedLen("id", 64)
+				}
+			})
```

Comment:
@Ronsor you mean alignment here for example? easiest is probably just an extra "ids_alignment" etc field after, other maybe more correct but less user friendly is to make "ids" into struct with "entries" and "alignment" i think

Reply:
Yes, there. I'll try both ways to see which is more usable.